homeassistant/components/sensor/openweathermap.py

```diff
@@ -27,10 +27,12 @@
     'rain': ['Rain', 'mm'],
     'snow': ['Snow', 'mm']
 }
+DEFAULT_NAME = "Openweathermap"
 
 PLATFORM_SCHEMA = vol.Schema({
     vol.Required(CONF_PLATFORM): 'openweathermap',
     vol.Required(CONF_API_KEY): vol.Coerce(str),
+    vol.Optional("name"): vol.Coerce(str),
```

Comment:
@emilhetty in case you're not familiar enough with Voluptuous, here's what @balloob is asking for:

Line 12 should be changed to:
```python
from homeassistant.const import (CONF_API_KEY, TEMP_CELSIUS, TEMP_FAHRENHEIT,
                                 CONF_PLATFORM, CONF_MONITORED_CONDITIONS,
                                 CONF_NAME)
```

Line 35 should be changed to:
```python
vol.Optional(CONF_NAME, default=DEFAULT_NAME): vol.Coerce(str),
```